velero-plugins/imagestream/shared.go

```diff
@@ -92,16 +95,42 @@ func GetRegistryEnvsForLocation(location string, namespace string) ([]string, er
 	if err != nil {
 		return nil, errors.New(fmt.Sprintf("errors getting registry env vars: %v", err))
 	}
-	return coreV1EnvVarArrToStringArr(envVars), nil
+	return coreV1EnvVarArrToStringArr(envVars, bsl.Namespace), nil
 }
 
-func coreV1EnvVarArrToStringArr(envVars []corev1.EnvVar) []string {
+func coreV1EnvVarArrToStringArr(envVars []corev1.EnvVar, namespace string) []string {
 	var envVarsStr []string
 	for _, envVar := range envVars {
-		envVarsStr = append(envVarsStr, coreV1EnvVarToString(envVar))
+		envVarsStr = append(envVarsStr, coreV1EnvVarToString(envVar, namespace))
 	}
 	return envVarsStr
 }
-func coreV1EnvVarToString(envVar corev1.EnvVar) string {
+func coreV1EnvVarToString(envVar corev1.EnvVar, namespace string) string {
+	if envVar.ValueFrom != nil {
+		if envVar.ValueFrom.SecretKeyRef != nil {	
```

Comment:
Could combine this with Line 109

Reply:
I was thinking maybe in the future secretkeyref will be one of many.. valueFrom used.. but rethinking now that's probably not gonna happen.. more likely will be refactoring to not create corev1.Envvar in the first place.
Will fix this.

Reply:
resolved in c3f4d94...14f229e